test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterTest.scala

```diff
@@ -383,6 +383,159 @@ class CharacterTest {
     assertFalse(Character.isDigit('\uFBFC'))
   }
 
+  @Test def toLowerCase_Char(): Unit = {
+    // Low
+    assertEquals('\n', Character.toLowerCase('\n'))
+    // Ascii
+    assertEquals('0', Character.toLowerCase('0'))
+    assertEquals('a', Character.toLowerCase('a'))
+    assertEquals('a', Character.toLowerCase('A'))
+    assertEquals('f', Character.toLowerCase('F'))
+    assertEquals('z', Character.toLowerCase('Z'))
+    // compat/ucd
+    assertEquals('β', Character.toLowerCase('β'))
+    assertEquals('θ', Character.toLowerCase('Θ'))
+    assertEquals('μ', Character.toLowerCase('μ'))
+    // alt (what is this?)
+    assertEquals('ā', Character.toLowerCase('ā'))
+    assertEquals('ā', Character.toLowerCase('Ā'))
+    assertEquals('ă', Character.toLowerCase('Ă'))
+    assertEquals('į', Character.toLowerCase('Į'))
+    // high
+    assertEquals('ａ', Character.toLowerCase('Ａ'))
+    assertEquals('ｚ', Character.toLowerCase('Ｚ'))
+    assertEquals('［', Character.toLowerCase('［'))
+  }
+
+  @Test def toLowerCase_Int(): Unit = {
+    // Low
+    assertEquals(10, Character.toLowerCase(10)) // '\n' => '\n'
+    // Ascii
+    assertEquals(97, Character.toLowerCase(97)) // a => a
+    assertEquals(97, Character.toLowerCase(65)) // A => a
+    assertEquals(102, Character.toLowerCase(70)) // F => f
+    assertEquals(122, Character.toLowerCase(90)) // Z => z
+    // compat/ucd
```

Comment:
Same issue with the comments. For `toLowerCase(int)`, there are additional interesting categories:

* Outside the BMP, not a letter (no change)
* Outside the BMP, lowercase (no change)
* (Outside the BMP, titlecase -- not sure that exists)
* Outside the BMP, uppercase

Reply:
Dumb question, what's BMP?

Reply:
[Basic Multilingual Plane](https://en.wikipedia.org/wiki/Plane_(Unicode)#Basic_Multilingual_Plane). It's the code points that fit in 1 UTF-16 code unit. That's opposed to the code points `>= 0x10000`, which require a surrogate pair.

Reply:
This is still pending. I'd still like to see tests broken down by categories that actually make sense. Combining what I wrote at https://github.com/scala-js/scala-js/pull/4207/commits/b9438ca40bccb2035cf31af613c459b559c8ded5#r494129429 and what is three comments above, a full list would be:

* ASCII repertoire, not a letter (no change)
* ASCII repertoire, lowercase letter (no change)
* ASCII repertoire, uppercase letter
* Somewhere else in the BMP, not a letter (no change)
* Somewhere else in the BMP, lowercase letter (no change)
* Somewhere else in the BMP, titlecase letter
* Somewhere else in the BMP, uppercase letter
* A high surrogate and a low surrogate (no change)
* Outside the BMP, not a letter (no change)
* Outside the BMP, lowercase (no change)
* (Outside the BMP, titlecase -- not sure that exists)
* Outside the BMP, uppercase

There can be just one of each of those.

Then the same would go for `toUpperCase(Int)` and `toTitleCase(Int)` (the exact same list of inputs can be reused, with different outputs). With that and the existing `toUpperCase_Int_special` and `toTitleCase_Int_special`, we'll have a good coverage.